CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ if(BUILD_SYCL_MODULE)
 else()
     option(GLIBCXX_USE_CXX11_ABI  "Set -D_GLIBCXX_USE_CXX11_ABI=1"           OFF)
 endif()
+option(ENABLE_SYCL_UNIFIED_SHARED_MEMORY "Enable SYCL unified shared memory" OFF)
```

Comment:
Will the use see an error if:
1. Both CUDA and SYCL flags are ON.
2. CUDA is OFF but CACHED_CUDA_MANAGER is ON.
3. SYCL is off but SYCL_UNIFIED_SHARED_MEMORY is ON.

Reply:
1. Added the check. 
2. It doesn't matter. CACHED_CUDA_MANAGER will have no effect in this case.
3. It doesn't matter. SYCL_UNIFIED_SHARED_MEMORY will have no effect in this case.